src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexRunner.cs

```diff
@@ -487,17 +483,31 @@ protected void EnsureStorage()
         /// </summary>
         protected bool IsBoundary(int index, int startpos, int endpos)
         {
-            return (index > startpos && RegexCharClass.IsBoundaryWordChar(runtext![index - 1])) !=
-                   (index < endpos && RegexCharClass.IsBoundaryWordChar(runtext![index]));
+            Debug.Assert(runtext != null, "runtext should not be null since this method is only callable by old codegen.");
```

Comment:
We can't really assert this.  IsBoundary is callable in a situation where runtext is null, even today, so asserting it's not null isn't valid.  We can just delete the assert.